pkg/controller/propagator/propagation.go

```diff
@@ -210,6 +201,89 @@ func (r *ReconcilePolicy) handleRootPolicy(instance *policiesv1.Policy) error {
 	return nil
 }
 
+// getApplicationPlacementDecisions return the placement decisions from an application
+// lifecycle placementrule
+func getApplicationPlacementDecisions(c client.Client, pb policiesv1.PlacementBinding, instance *policiesv1.Policy) ([]appsv1.PlacementDecision, *policiesv1.Placement, error) {
+	plr := &appsv1.PlacementRule{}
+	err := c.Get(context.TODO(), types.NamespacedName{Namespace: instance.GetNamespace(),
+		Name: pb.PlacementRef.Name}, plr)
+	if err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "Failed to get plr...", "Namespace", instance.GetNamespace(), "Name",
+			pb.PlacementRef.Name)
+		return nil, nil, err
+	}
+	// plr found, add current plcmnt to placement
+	placement := &policiesv1.Placement{
+		PlacementBinding: pb.GetName(),
+		PlacementRule:    plr.GetName(),
+		// Decisions:        plr.Status.Decisions,
+	}
+	return plr.Status.Decisions, placement, nil
+}
+
+// getClusterPlacementDecisions return the placement decisions from cluster
+// placement decisions
+func getClusterPlacementDecisions(c client.Client, pb policiesv1.PlacementBinding, instance *policiesv1.Policy) ([]appsv1.PlacementDecision, *policiesv1.Placement, error) {
+	plr := &clusterv1alpha1.Placement{}
+	err := c.Get(context.TODO(), types.NamespacedName{Namespace: instance.GetNamespace(),
+		Name: pb.PlacementRef.Name}, plr)
+	if err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "Failed to get plr...", "Namespace", instance.GetNamespace(), "Name",
+			pb.PlacementRef.Name)
+		return nil, nil, err
+	}
+	// plr found, add current plcmnt to placement
+	placement := &policiesv1.Placement{
+		PlacementBinding: pb.GetName(),
+		Placement:        plr.GetName(),
+		// Decisions:        plr.Status.Decisions,
+	}
+	list := &clusterv1alpha1.PlacementDecisionList{}
+	lopts := &client.ListOptions{Namespace: instance.GetNamespace()}
+
+	opts := client.MatchingLabels{"cluster.open-cluster-management.io/placement": plr.GetName()}
+	opts.ApplyToList(lopts)
+	err = c.List(context.TODO(), list, lopts)
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
I'm curious, have you verified if `List` returns a not found error if the API resource returned nothing? Normally, a REST API would not give you a 404 for instance. It'd just return an empty list, so this wasn't what I was expected is all.

Reply:
right, not sure about this